main.c

```diff
@@ -40,23 +39,22 @@ int main(int argc, __attribute__((unused)) char **argv, char **env) {
 
 		if (str_cmp(exec_argv[0], "exit") == 0)
 			break;
-
-		if ((pid = fork()) == -1) {
+		if (str_cmp(exec_argv[0], "cd") == 0) {
+			ch_dir(exec_argv[1], env);
+		} else if ((pid = fork()) == -1) {
 			perror("fork");
 			return 1;
 		} else if (pid == 0) {
-			execve(raw_str = concat_strings("/bin/", exec_argv[0]), exec_argv, env);
+			execve(raw_str = find_path(exec_argv[0], env), exec_argv, env);
```

Comment:
I believe valgrind throws a warning because find_path doesn't always return a string (specifically, if the command is not found). One surefire way to squelch this warning would be to check first if the file exists.